src/inmanta/export.py

```diff
@@ -469,14 +478,26 @@ def add_resource(self, resource: Resource) -> None:
 
         self._resources[resource.id] = resource
 
-    def resources_to_list(self) -> list[dict[str, Any]]:
-        """Convert the resource list to a json representation"""
+    def resources_to_list(self) -> tuple[list[dict[str, Any]], set[str]]:
+        """
+        Convert the resource list to a json representation
+        If the soft_delete option is set, this will also compute the set of resource sets containing these resources
```

Comment:
I think this could be a lot simpler if you put it in `_load_resource_sets`, or am I missing something. We could even go so far as to add an attribute `self._resource_sets: set[str]` because that method already works by populating an attribute rather than returning its result.